src/voodoo-prep/prep.ml

```diff
@@ -12,10 +12,32 @@ let process_package : Fpath.t -> Package.t -> Fpath.t list -> unit =
  fun root package files ->
   let dest = Package.prep_path package in
 
+  let format_matches_compiler_version path =
+    let open Unix in
+    let cmd = "ocamlobjinfo " ^ Fpath.to_string path in
+    let ic, oc = open_process cmd in
+    let rec matches_version ic =
+      try
+        let line = input_line ic in
+        let prefix = "Wrong magic number" in
+        let str_len = String.length line in
+        if str_len >= 18 && String.sub line 0 18 = prefix then (
+          Printf.eprintf "ERROR: Ocamlobjinfo on %s: %s\n"
+            (Fpath.to_string path) line;
+          false)
+        else if str_len > 0 then matches_version ic
+        else true
+      with End_of_file -> true
+    in
+    let matches_version = matches_version ic in
+    let _ = close_process (ic, oc) in
+    matches_version
+  in
+
   (* Some packages produce ocaml artefacts that can't be processed with the switch's
       ocaml compiler - most notably the secondary compiler! This switch is intended to
       be used to ignore those files *)
-  let process_ocaml_artefacts =
+  let is_not_blacklisted =
```

Comment:
Personally I'm not a fan of `not_...` variables, I think this part would be much clearer if it was `is_blacklisted` and we test its negation everywhere.